File: Code/client/src/routes/plans/models/planPaymentGridItemModel.ts
export interface PlanPaymentGridItemModel {
    planPaymentId: number;
    paymentStartDate: string;
    paymentEndDate?: string;
    companyName?: string;
    description?: string;
    categoryName?: string;
    walletName?: string;
    personName?: string;
    valueFormat: string;
    value: number;
    taxable: boolean;
    taxYear?: number;
    isActive: boolean;
    isActiveAndInDate: boolean;
}

File: Code/SimpleBudget.API/Models/PlanPayments/PlanPaymentGridItemModel.cs
namespace SimpleBudget.API.Models
{
    public class PlanPaymentGridItemModel
    {
        public int PlanPaymentId { get; set; }
        public string PaymentStartDate { get; set; } = default!;
        public string? PaymentEndDate { get; set; }
        public string? CompanyName { get; set; }
        public string? Description { get; set; }
        public string? CategoryName { get; set; }
        public string? WalletName { get; set; }
        public string? PersonName { get; set; }
        public string ValueFormat { get; set; } = default!;
        public decimal Value { get; set; }
        public bool Taxable { get; set; }
        public int? TaxYear { get; set; }
        public bool IsActive { get; set; }
        public bool IsActiveAndInDate { get; set; }
    }
}
